### DISCUSSION TITLE
insert_textbot render a question mark for the uero sign (€)

### DISCUSSION THREAD

**Discussion by Bogghi** [pymupdf/PyMuPDF#1978] [Q&A] [answered]
I've been working on populating a template with data we have on a db.

I've got no problem placing text on the template but when I try to add a euro sign in the text it renders a question mark, could it be related to the character missing from the font set? If yes, how should I proceed from here? What could be the cause of this situation if it's not the font?

Here is the code I've written, and it's giving me issue:

```
#template is the path of the pdf file
doc = fitz.open(template)

shape = doc[5].new_shape()
textRect = fitz.Rect(
    280, 230,
    380, 260
)
shape.draw_rect(textRect)
#self.row is the result of a query from db
shape.insert_textbox(textRect, f"€ {self.row[37]}", fontsize = 16, color = (0,0,0), align = 2)
shape.finish(width = 0.3, color=(1,1,1))
shape.commit()

doc.save(f"{self.currentFlierFolder}/{self.filename}.pdf", garbage=4, deflate=True, clean=True)
```

Thanks a lot to anyone who will help on this topic

**Comment by Bogghi** [comment]
![image](https://user-images.githubusercontent.com/11005947/196920632-a1e223f9-5366-42f5-972d-4327296b9f65.png)

here's how I see it in the PDF

**Reply by JorjMcKie** [reply]
If using `page.insert_text()` with default fonts (i.e. Base-14 Helvetica) then only characters with unicode number < 256 are available.
The € sign is contained in it (`0x80`), but Python understands € as `hex(ord("€")) = '0x20ac'`.
So you have 2 choices:
1. Replace € by `0x80` in your text and see if this solves it
2. Use a better font like one of `pymupdf-fonts` or one of your system fonts

**Comment by JorjMcKie** [comment, marked as answer]
this works:
```python
import fitz
text = "Total amount is € 32.368,00"
doc=fitz.open()
page=doc.new_page()
page.insert_text((72,72), text)
page.insert_text((72,92), text.replace("€",chr(0x80)))
doc.save("x.pdf")
```
![grafik](https://user-images.githubusercontent.com/8290722/196947472-2c067302-8f6d-4437-9ed5-48f46409ee85.png)

**Reply by Bogghi** [reply]
Thank you so much dude, you saved my day
